tools/protovalidate-conformance/internal/cases/cases_ignore_empty.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cases
+
+import (
+	"github.com/bufbuild/protovalidate/tools/internal/gen/buf/validate"
+	"github.com/bufbuild/protovalidate/tools/internal/gen/buf/validate/conformance/cases"
+	"github.com/bufbuild/protovalidate/tools/protovalidate-conformance/internal/results"
+	"github.com/bufbuild/protovalidate/tools/protovalidate-conformance/internal/suites"
+	"google.golang.org/protobuf/proto"
+)
+
+func ignoreEmptySuite() suites.Suite {
+	return suites.Suite{
+		"proto2/scalar/optional/nonzero": suites.Case{
+			Message:  &cases.IgnoreEmptyProto2ScalarOptional{Val: proto.Int32(42)},
+			Expected: results.Success(true),
+		},
+		"proto2/scalar/optional/zero": suites.Case{
+			Message:  &cases.IgnoreEmptyProto2ScalarOptional{Val: proto.Int32(0)},
+			Expected: results.Violations(&validate.Violation{FieldPath: "val", ConstraintId: "int32.gt"}),
+		},
+		"proto2/scalar/optional/unset": suites.Case{
+			Message:  &cases.IgnoreEmptyProto2ScalarOptional{},
+			Expected: results.Success(true),
+		},
+		"proto2/scalar/optional_with_default/nonzero": suites.Case{
+			Message:  &cases.IgnoreEmptyProto2ScalarOptionalWithDefault{Val: proto.Int32(1)},
+			Expected: results.Success(true),
+		},
+		"proto2/scalar/optional_with_default/zero": suites.Case{
+			Message:  &cases.IgnoreEmptyProto2ScalarOptionalWithDefault{Val: proto.Int32(0)},
+			Expected: results.Violations(&validate.Violation{FieldPath: "val", ConstraintId: "int32.gt"}),
+		},
+		"proto2/scalar/optional_with_default/default": suites.Case{
+			Message:  &cases.IgnoreEmptyProto2ScalarOptionalWithDefault{Val: proto.Int32(42)},
+			Expected: results.Success(true),
```

Comment:
Since `optional` is nullable, it's a noop for this rule regardless. There's a "nonzero" rule above that is neither the default and not zero that captures that case. I will add tests with a negative value which I think is what will cover the remaining missing range of behavior.

Reply:
Said another way setting a default doesn't matter for this rule.